### PR TITLE
refactor(text-editor): covert tests to use react-testing-library

### PR DESCRIPTION
**Pull request by robinzigmond**
### Proposed behaviour

TextEditor component should have unit tests written in React Testing Library

NOTE: due to many issues with the drafjs library and jsdom/RTL, a lot of functionality was not possible to sensibly test, so these tests have been added to Playwright instead.  Many unit tests have been added just for coverage that do not otherwise provide much value. When we eventually move away from draftjs (which is no longer supported) we should write RTL tests for what is being covered by the new Playwright tests, and remove both those playwright tests and as many as possible of the current "coverage only" unit tests.

### Current behaviour

Unit tests use Enzyme

### Checklist

- [X] Commits follow our style guide
- [X] Related issues linked in commit messages if required
- [X] Screenshots are included in the PR if useful
- [X] All themes are supported if required
- [X] Unit tests added or updated if required
- [X] Playwright automation tests added or updated if required
- [X] Storybook added or updated if required
- [X] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [X] Typescript `d.ts` file added or updated if required
- [X] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/text-editor/text-editor.test.tsx [hunk 1]
```diff
@@ -0,0 +1,946 @@
...
+  );
+
+  expect(
+    screen.getByRole("textbox", { name: "Text Editor Label" })
+  ).toHaveAccessibleDescription(
+    "here is a hint You can enter up to 10 characters"
+  );
+});
+
+test("files a console warning when the `rows` prop is passed as a number less than 2", () => {
```
Comment (DipperTheDan) [inline src/components/text-editor/text-editor.test.tsx:416]: **suggestion/question/nitpick:** Is this a spelling mistake?  

```suggestion
test("fires a console warning when the `rows` prop is passed as a number less than 2", () => {
```

File: src/components/text-editor/text-editor.test.tsx [hunk 2]
```diff
@@ -0,0 +1,946 @@
...
+
+  await user.click(screen.getByRole("button", { name: "italic" }));
+  await user.type(
+    screen.getByRole("textbox", { name: "Text Editor Label" }),
+    "italic text"
+  );
+  expect(screen.getByText("talic texti")).toBeVisible();
+}, 10000); // allow to run for up to 10 seconds due to timeout issues
+
+// for coverage only - this behaviour doesn't work properly in jsdom (note incorrect order of text being entere).
```
Comment (DipperTheDan) [inline src/components/text-editor/text-editor.test.tsx:467]: **nitpick:**

```suggestion
// for coverage only - this behaviour doesn't work properly in jsdom (note incorrect order of text being entered).
```

File: src/components/text-editor/text-editor.test.tsx [hunk 3]
```diff
@@ -0,0 +1,946 @@
...
+
+  await user.click(screen.getByRole("button", { name: "bullet-list" }));
+  await user.type(
+    screen.getByRole("textbox", { name: "Text Editor Label" }),
+    "list item"
+  );
+  expect(screen.getByRole("listitem")).toHaveTextContent("ist iteml");
+}, 10000); // allow to run for up to 10 seconds due to timeout issues
+
+// for coverage only - this behaviour doesn't work properly in jsdom (note incorrect order of text being entere).
```
Comment (DipperTheDan) [inline src/components/text-editor/text-editor.test.tsx:481]: **nitpick:**

```suggestion
// for coverage only - this behaviour doesn't work properly in jsdom (note incorrect order of text being entered).
```

File: src/components/text-editor/text-editor.test.tsx [hunk 4]
```diff
@@ -0,0 +1,946 @@
...
+  await user.type(
+    screen.getByRole("textbox", { name: "Text Editor Label" }),
+    "bold text"
+  );
+
+  expect(screen.getByText("old textb")).toBeVisible();
+});
+
+// the next several tests are for coverage only - due to an issue with React itself, dispatching the (deprecated) `keypress` event
+// with non-standard propertoes is the only way to trigger the `handleBeforeInput` handler in a test. See
```
Comment (DipperTheDan) [inline src/components/text-editor/text-editor.test.tsx:576]: **nitpick:**

```suggestion
// with non-standard properties is the only way to trigger the `handleBeforeInput` handler in a test. See
```

Comment (edleeks87) [reply src/components/text-editor/text-editor.test.tsx:576]: image

**Review by DipperTheDan — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

File: src/components/text-editor/text-editor.pw.tsx [hunk 5]
```diff
@@ -789,3 +836,531 @@ test("should set aria-live attribute on Character Count to `polite` when compone
...
+    // copy text to clipboard
+    await page.evaluate(() => navigator.clipboard.writeText("1234567890"));
+
+    if (process.env.CI) {
+      // assume Ubuntu (which is the OS the Github actions run on), where Control+Shift+V is needed to paste
+      await page.keyboard.down("Control");
+      await page.keyboard.down("Shift");
+      await page.keyboard.press("v");
+      await page.keyboard.up("Shift");
+      await page.keyboard.up("Control");
```
Comment (stephenogorman) [inline src/components/text-editor/text-editor.pw.tsx:1325]: Could you add the following for anyone who might be running tests locally on Windows?
```
} else if (process.platform === "win32") {
  // Windows: Control+V
  await page.keyboard.down("Control");
  await page.keyboard.press("v");
  await page.keyboard.up("Control");

```

File: src/components/text-editor/text-editor.pw.tsx [hunk 6]
```diff
@@ -789,3 +836,531 @@ test("should set aria-live attribute on Character Count to `polite` when compone
...
+      navigator.clipboard.writeText("12345678901234567890")
+    );
+
+    if (process.env.CI) {
+      // assume Ubuntu (which is the OS the Github actions run on), where Control+Shift+V is needed to paste
+      await page.keyboard.down("Control");
+      await page.keyboard.down("Shift");
+      await page.keyboard.press("v");
+      await page.keyboard.up("Shift");
+      await page.keyboard.up("Control");
```
Comment (stephenogorman) [inline src/components/text-editor/text-editor.pw.tsx:1356]: Same as previous comment, could you add the following for anyone who might be running tests locally on Windows?
```
} else if (process.platform === "win32") {
  // Windows: Control+V
  await page.keyboard.down("Control");
  await page.keyboard.press("v");
  await page.keyboard.up("Control");
```

File: src/components/text-editor/text-editor.pw.tsx (hunk 5 above)
Comment (robinzigmond) [reply src/components/text-editor/text-editor.pw.tsx:1325]: that's a great point, thanks for raising it and I'm happy to do this.

In fact, I wasn't even aware of `process.platform`, that looks like what I should have been using all along here so I'll change the other `if` condition to use this (assuming it's easy to test for Linux/Ubuntu).

**Review by stephenogorman — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**